functests/performance/updating_profile.go

```diff
@@ -43,80 +47,91 @@ var _ = Describe("[rfe_id:28761] Updating parameters in performance profile", fu
-		waitForMcpCondition(testutils.RoleWorkerRT, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue, timeout)
-
-		By("Waiting when mcp finishes updates and verifying new parameters applied")
-		waitForMcpCondition(testutils.RoleWorkerRT, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue, timeout)
+			By("Modifying profile")
+			profile.Spec.HugePages = &performancev1alpha1.HugePages{
+				DefaultHugePagesSize: &hpSize,
+				Pages: []performancev1alpha1.HugePage{
+					{
+						Count: 5,
+						Size:  hpSize,
+					},
+				},
+			}
+			profile.Spec.CPU = &performancev1alpha1.CPU{
+				BalanceIsolated: &f,
+				Reserved:        &reserved,
+				Isolated:        &isolated,
+			}
+			profile.Spec.NUMA = &performancev1alpha1.NUMA{
+				TopologyPolicy: &policy,
+			}
+			profile.Spec.RealTimeKernel = &performancev1alpha1.RealTimeKernel{
+				Enabled: &f,
+			}
 
-		for _, node := range workerRTNodes {
-			checkCmdLineArgs(node, newCmdLineArgs)
+			if profile.Spec.AdditionalKernelArgs == nil {
+				By("AdditionalKernelArgs is empty. Checking only adding new arguments")
+				profile.Spec.AdditionalKernelArgs = append(profile.Spec.AdditionalKernelArgs, "new-argument=test")
+			} else {
+				removedKernelArgs = profile.Spec.AdditionalKernelArgs[0]
+				profile.Spec.AdditionalKernelArgs = append(profile.Spec.AdditionalKernelArgs[1:], "new-argument=test")
+			}
 
-			cmd := []string{"cat", "/rootfs/etc/kubernetes/kubelet.conf"}
-			Expect(execCommandOnWorker(cmd, &node)).To(ContainSubstring(fmt.Sprintf(`"reservedSystemCPUs":"%s"`, reserved)))
-			Expect(execCommandOnWorker(cmd, &node)).To(ContainSubstring(fmt.Sprintf(`"topologyManagerPolicy":"%s"`, policy)))
+			By("Verifying that mcp is ready for update")
+			waitForMcpCondition(testutils.RoleWorkerRT, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue, timeout)
+
+			By("Applying changes in performance profile and waiting until mcp will start updating")
+			Expect(testclient.Client.Update(context.TODO(), profile)).ToNot(HaveOccurred())
+			waitForMcpCondition(testutils.RoleWorkerRT, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue, timeout)
+
+			By("Waiting when mcp finishes updates")
+			waitForMcpCondition(testutils.RoleWorkerRT, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue, timeout)
+		})
+
+		table.DescribeTable("Verify that profile parameters were updated", func(cmd, parameter []string, shouldContain bool) {
+			for _, node := range workerRTNodes {
+				for _, param := range parameter {
+					if shouldContain {
+						Expect(execCommandOnWorker(cmd, &node)).To(ContainSubstring(param))
+					} else {
+						Expect(execCommandOnWorker(cmd, &node)).NotTo(ContainSubstring(param))
+					}
+				}
+			}
+		},
+			table.Entry("[test_id:28024] verify that hugepages size and count updated", chkCmdLine, []string{"default_hugepagesz=2M", "hugepagesz=2M", "hugepages=5"}, true),
```

Comment:
do you assume that you always will have the same order of tests under the spec?

Reply:
it's a list of arguments, so I don't care about order. Here I'm going thru the list:

https://github.com/openshift-kni/performance-addon-operators/blob/3187fb61509fc378e66237be23338d4d89742883/functests/performance/updating_profile.go#L106

Reply:
hm, I'm not 100% sure, but as I see tests under one "Describe" goes one by one.. at least i didn't see any issues with order after several running..

Reply:
In general yes, see - https://onsi.github.io/ginkgo/#spec-permutation, but it bad practice to assume it, and ginkgo gives the way to run it fully randomize, and it also recommended to run this way under the CI.

Reply:
I tried to find the way how to do all "updating" stuff before running any It() in this file, but unfortunately i didn't... 

Do you have any idea?